### PR TITLE
[FE] Create a Button Molecule with dropdown button options

### PR DESCRIPTION
**Pull request by b-l-i-n-d**

### CODE REVIEW

File: assets/react/v3/shared/molecules/DropdownButton.tsx [hunk 1]
```diff
@@ -0,0 +1,442 @@
...
+  type = 'button',
+  text,
+  children,
+  variant = 'primary',
+  size = 'medium',
+  icon,
+  iconPosition = 'left',
+  loading = false,
+  disabled = false,
+  tabIndex,
```
Comment (ahamed) [inline assets/react/v3/shared/molecules/DropdownButton.tsx:55]:
```suggestion
  tabIndex = -1,
```

File: assets/react/v3/shared/molecules/DropdownButton.tsx [hunk 2]
```diff
@@ -0,0 +1,442 @@
...
+          )}
+          <span css={[styles.buttonContent(loading, disabled), buttonContentCss]}>
+            {icon && iconPosition === 'left' && <span css={styles.buttonIcon(iconPosition)}>{icon}</span>}
+            {text}
+            {icon && iconPosition === 'right' && <span css={styles.buttonIcon(iconPosition)}>{icon}</span>}
+          </span>
+        </button>
+        <button
+          ref={dropdownTriggerRef}
+          type={'button'}
```
Comment (ahamed) [inline assets/react/v3/shared/molecules/DropdownButton.tsx:86]:
```suggestion
          type='button'
```

File: assets/react/v3/shared/molecules/DropdownButton.tsx [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+          css={[styles.button(variant, size, (loading = false), disabled), styles.dropdownButton(variant, disabled)]}
```
Comment (ahamed) [inline assets/react/v3/shared/molecules/DropdownButton.tsx:87]: Could you please explain this `(loading = false)` thing?

File: assets/react/v3/shared/molecules/DropdownButton.tsx [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+        isOpen={isOpen}
+        closePopover={() => setIsOpen(false)}
+        animationType={AnimationType.slideUp}
+      >
+        <div css={styles.dropdownWrapper}>
+          {React.Children.map(children, child => {
+            if (React.isValidElement(child)) {
+              const childProps: DropdownOptionProps = {
+                ...child.props,
+                onClick: (e: MouseEvent<HTMLButtonElement>) => {
```
Comment (ahamed) [inline assets/react/v3/shared/molecules/DropdownButton.tsx:107]:
```suggestion
                onClick: (event: MouseEvent<HTMLButtonElement>) => {
```

File: assets/react/v3/shared/molecules/DropdownButton.tsx [hunk 5]
```diff
@@ -0,0 +1,442 @@
...
+  wrapper: css`
+    ${styleUtils.display.inlineFlex()};
+    align-items: center;
+  `,
+  button: (variant: ButtonVariant, size: ButtonSize, loading: boolean, disabled: boolean) => css`
+    ${styleUtils.resetButton};
+    display: inline-block;
+    font-size: ${fontSize[15]};
+    line-height: ${lineHeight[24]};
+    font-weight: ${fontWeight.medium};
```
Comment (ahamed) [inline assets/react/v3/shared/molecules/DropdownButton.tsx:146]: It should come from a typography.

**Review by ahamed — CHANGES_REQUESTED** [review]

File: assets/react/v3/shared/molecules/DropdownButton.tsx (hunk 3 above)
Comment (b-l-i-n-d) [reply assets/react/v3/shared/molecules/DropdownButton.tsx:87]: Both of them use the same style so both can have the same style, but we don't need the dropdown trigger button to have a loading state. We are receiving loading props for the main button. To make sure it does not have any effect from loading props, it is defaulted as false.

**Review by ahamed — APPROVED** [review]

**Pull request merged**